docs/pages/api.js

```diff
@@ -0,0 +1,755 @@
+      );
+    case 'enum':
+      return (
+        <Join
+          separator={
+            <React.Fragment>
+              <br />
+              |&nbsp;
+            </React.Fragment>
+          }
+        >
+          {type.value.map(member => {
+            return member.value;
+          })}
+        </Join>
+      );
+    case 'union':
+      return (
+        <Join
+          separator={
+            <React.Fragment>
+              <br />
+              |&nbsp;
+            </React.Fragment>
+          }
+        >
+          {type.value.map((member, index) => {
+            return <PropType key={`member-${index}`} type={member} />;
+          })}
+        </Join>
+      );
+    case 'arrayOf':
+      return (
+        <React.Fragment>
+          Array{'<'}
+          <PropType type={type.value} />
+          {'>'}
+        </React.Fragment>
+      );
+    case 'instanceOf':
+      if (type.value.startsWith('typeof')) {
+        return /typeof (.*) ===/.exec(type.value)[1];
+      }
+      return type.value;
+    default:
+      return type.name || 'unknown';
+  }
+}
+
+PropType.propTypes = { type: PropTypes.object.isRequired };
+
+const useComponentPropStyles = makeStyles(theme => {
+  // let the specificity games begin
+  return {
+    propDefault: {
+      fontSize: 13,
+      fontFamily: 'Consolas, "Liberation Mono", Menlo, monospace',
+      borderBottom: `1px dotted ${theme.palette.divider}`,
+    },
+    propName: {
+      'table td&': {
+        fontSize: 13,
+        fontFamily: 'Consolas, "Liberation Mono", Menlo, monospace',
+      },
+    },
+    propType: {
+      'table td&': {
+        fontSize: 13,
+        fontFamily: 'Consolas, "Liberation Mono", Menlo, monospace',
+        color: theme.palette.type === 'light' ? '#932981' : '#ffb6ec',
+      },
+    },
+    required: {
+      'table td&': {
+        color: theme.palette.type === 'light' ? '#006500' : '#a5ffa5',
+      },
+    },
+  };
+});
+
+function ComponentProp(props) {
+  const { name, prop } = props;
+
+  // injected by withStyles
+  const required = name === 'classes' ? false : prop.required;
+
+  const { description, tags, type } = prop;
+  const defaultValue = prop.defaultValue ? prop.defaultValue.value.replace(/\r*\n/g, '') : null;
+
+  const classes = useComponentPropStyles();
+
+  return (
+    <tr>
+      <td className={clsx(classes.propName, { [classes.required]: required })}>
+        {name}
+        {required && '&nbsp;*'}
+      </td>
+      <td className={classes.propType}>
+        <PropType type={type} />
+      </td>
+      <td>
+        <span className={classes.propDefault}>{defaultValue}</span>
+      </td>
+      <td>
+        <PropDescription description={description} tags={tags} type={type} />
+      </td>
+    </tr>
+  );
+}
+
+ComponentProp.propTypes = { name: PropTypes.string.isRequired, prop: PropTypes.object.isRequired };
+
+function ComponentPropsTable(props) {
+  const { propsApi } = props;
+
+  return (
+    <table>
```

Comment:
Have you considered to 1. use the Table components, as in https://material-ui-pickers.dev/api/KeyboardDatePicker. Or 2. to use the MarkdownElement components?

Reply:
There's no need to use our Table components since we probably reset 99% of the styles. The MarkdownElement is for markdown not for tables, right?

Reply:
Yeah, I think that generating a table markdown string would be too limited for our use case. I would be more in favor of using our Table component here and make changes to the table Markdown version if it doesn't match.